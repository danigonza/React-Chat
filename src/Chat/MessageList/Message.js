import React from 'react'

const getClassName = (message) =>
    message.from_id === 1 ? 'self' : 'other'

const Message = ({message}) => <li className={getClassName(message)}><div className='message'><p>{message.body}</p></div></li>

export default Message
