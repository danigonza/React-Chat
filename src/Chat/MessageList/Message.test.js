import React from 'react'
import Message from './Message'
import renderer from 'react-test-renderer'

it('renders correctly', () => {
  const message = { id: 1, room_id: 1, body: 'Uola', from_id: 1 }
  const tree = renderer.create(
    <Message message={message} />
    ).toJSON()
  expect(tree).toMatchSnapshot()
})
