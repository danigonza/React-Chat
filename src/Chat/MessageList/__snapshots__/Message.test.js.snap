// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<li
  className="self"
>
  <div
    className="message"
  >
    <p>
      Uola
    </p>
  </div>
</li>
`;
